src/com/owncloud/android/operations/SynchronizeFileOperation.java

```diff
@@ -241,10 +243,18 @@ private void requestForUpload(OCFile file) {
      * @param file OCFile object representing the file to download
      */
     private void requestForDownload(OCFile file) {
-        Intent i = new Intent(mContext, FileDownloader.class);
-        i.putExtra(FileDownloader.EXTRA_ACCOUNT, mAccount);
-        i.putExtra(FileDownloader.EXTRA_FILE, file);
-        mContext.startService(i);
+        Intent intent = new Intent(mContext, FileDownloader.class);
+        intent.putExtra(FileDownloader.KEY_ACCOUNT, mAccount);
+        intent.putExtra(FileDownloader.KEY_FILE, file);
+
+        // Since in Android O and above the apps in background are not allowed to start background
+        // services and available offline feature may try to do it, this is the way to proceed
+        if (mContext instanceof AvailableOfflineSyncJobService && Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
The same as before. Use ContextCompat


    if (context instanceof AvailableOfflineSyncJobService && Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
        intent.putExtra(FileUploader.KEY_IS_AVAILABLE_OFFLINE_FILE, true);
    }
    ContextCompat.startForegroundService(context, intent)


BTW, Can this code be refactored so you only have a common section?